stdlib/list.gr

```diff
@@ -46,42 +95,122 @@ let reverse = list => {
   iter(list, [])
 }
 
-let rec append = (list1, list2) => {
+/**
+ * Creates a new list with the elements the first list, followed by
+ * the elements of the second list. This does not modify the arguments.
+ *
+ * @param list1: The list containing elements to appear first
+ * @param list2: The list containing elements to appear second
+ * @returns The new list containing elements from `list1` followed by elements from `list2`
+ *
+ * @since v0.1.0
+ */
+export let rec append = (list1, list2) => {
   match (list1) {
     [] => list2,
     [first, ...rest] => [first, ...append(rest, list2)],
   }
 }
 
-let rec contains = (value, list) => {
+/**
+ * Checks if the value is an element of the input list.
+ * Uses the generic `==` structural equality operator.
+ *
+ * @param search: The value to compare
+ * @param list: The list to inspect
+ * @returns `true` if the value exists in the list or `false` otherwise
+ *
+ * @since v0.1.0
+ */
+export let rec contains = (search, list) => {
   match (list) {
     [] => false,
-    [first, ...rest] => first == value || contains(value, rest),
+    [first, ...rest] => first == search || contains(search, rest),
   }
 }
 
-let rec reduce = (fn, acc, list) => {
+/**
+ * Combines all elements of a list using a reducer function,
+ * starting from the "head", or left side, of the list.
+ *
+ * In `List.reduce(fn, initial, list)`, `fn` is called with
+ * an accumulator and each element of the list, and returns
+ * a new accumulator. The final value is the last accumulator
+ * returned. The accumulator starts with value `initial`.
+ *
+ * @param fn: The reducer function to call on each element, where the value returned will be the next accumulator value
+ * @param initial: The initial value to use for the accumulator on the first iteration
+ * @param list: The list to iterate
+ * @returns The final accumulator returned from `fn`
+ *
+ * @example List.reduce((a, b) => a + b, 0, [1, 2, 3]) // 6
+ *
+ * @since v0.2.0
+ * @history v0.1.0: Originally named `foldLeft`
+ * @history v0.2.0: Renamed to `reduce`
+ */
+export let rec reduce = (fn, initial, list) => {
   match (list) {
-    [] => acc,
-    [first, ...rest] => reduce(fn, fn(acc, first), rest),
+    [] => initial,
```

Comment:
Maybe just me but acc for accumulator makes sense, initial makes me think it's the initial (unchanging) value

Reply:
Changed my mind, this makes the documentation look much clearer